app/viewmodels/yourAccount/ReturnsViewModel.scala

```diff
@@ -17,34 +17,80 @@
 package viewmodels.yourAccount
 
 import models.StandardPeriod
-import models.SubmissionStatus.{Due, Next, Overdue}
 import pages.{EmptyWaypoints, Waypoints}
 import play.api.i18n.Messages
 import viewmodels.{LinkModel, Paragraph, ParagraphSimple, ParagraphWithId}
 
 import java.time.format.DateTimeFormatter
 
 case class ReturnsViewModel(
-                           contents: Seq[Paragraph],
-                           linkToStart: Option[LinkModel] = None
+                             contents: Seq[Paragraph],
+                             linkToStart: Option[LinkModel] = None
                            )
 
 object ReturnsViewModel {
 
   def apply(returns: Seq[Return])(implicit messages: Messages): ReturnsViewModel = {
-    val inProgress = returns.find(_.inProgress)
-    val returnDue = returns.find(_.submissionStatus == Due)
-    val overdueReturns = returns.filter(_.submissionStatus == Overdue)
-    val nextReturn = returns.find(_.submissionStatus == Next)
+    val runtimeExceptionOrMaybeActionableReturn = NextReturnCalculation.calculateNonNextReturn(returns)
+    val waypoints = EmptyWaypoints
+    runtimeExceptionOrMaybeActionableReturn match {
+      case Left(runtimeException) => throw runtimeException
+      case Right(actionableReturn) =>
+        actionableReturn match {
+          case NextReturn(pendingReturn) =>
+            ReturnsViewModel(
+              contents = Seq(nextReturnNoLinkParagraph(pendingReturn.period))
+            )
+
+          case otherReturn: OtherReturn =>
+            createModelFromDueReturn(waypoints, otherReturn)
+        }
+    }
+  }
+
+  private def nextReturnNoLinkParagraph(nextReturn: StandardPeriod)(implicit messages: Messages) =
```

Comment:
Moved from bottom and NoLink added to name as this should not be a valid return for doing in the filter.